BridgeCareApp/BridgeCare/DataAccessLayer/InvestmentStrategies.cs

```diff
@@ -18,25 +21,101 @@ public InvestmentStrategies(BridgeCareContext context)
                     {
-                        Id = p.SIMULATIONID,
-                        BudgetOrder = p.BUDGETORDER,
-                        FirstYear = p.FIRSTYEAR ?? 0,
-                        NumberYears = p.NUMBERYEARS ?? 0,
-                        InflationRate = p.INFLATIONRATE ?? 0,
-                        DiscountRate = p.DISCOUNTRATE ?? 0
-                    });
+                        Name = p.SIMULATION1,
+                        Description = p.COMMENTS,
+                        SimulationId = p.SIMULATIONID,
+                        NetworkId = p.NETWORKID ?? 0,
+                        FirstYear = p.INVESTMENTS.FIRSTYEAR ?? 0,
+                        NumberYears = p.INVESTMENTS.NUMBERYEARS ?? 0,
+                        InflationRate = p.INVESTMENTS.INFLATIONRATE ?? 0,
+                        DiscountRate = p.INVESTMENTS.DISCOUNTRATE ?? 0,
+                        BudgetOrder = p.INVESTMENTS.BUDGETORDER,
+                        YearlyBudget = p.YEARLYINVESTMENTs.Select(m => new InvestmentStrategyYearlyBudgetModel
+                        {
+                            Year = m.YEAR_,
+                            Budget = p.YEARLYINVESTMENTs
+                            .Where(n => n.YEAR_ == m.YEAR_)
+                            .Select(f => new InvestmentStrategyBudgetModel()
+                            {
+                                Amount = f.AMOUNT,
+                                Name = f.BUDGETNAME
+                            }).ToList()
+                        }).ToList()
+                    }).ToList();
+
+                //Cannot do this inside the linq query, would be nice
+                foreach (InvestmentStrategyModel model in simulation)
+                {
+                    model.SetBudgets();
+                }
+
+                return simulation.AsQueryable();
+            }
+            catch (SqlException ex)
+            {
+                HandleException.SqlError(ex, "Investment Strategies");
+            }
+            return Enumerable.Empty<InvestmentStrategyModel>().AsQueryable();
+        }
+
+        public bool SetInvestmentStrategies(InvestmentStrategyModel data, BridgeCareContext db)
+        {
+            //Ensures budget order is transferred from array storage as it comes in from json to
+            //the databse format, comma delimited
+            string budgetOrder = data.GetBudgetOrder();
+
+            //Derive FirstYear and NumberYears from the YearlyBudget list.
+            data.FirstYear = data.YearlyBudget.Min(r => r.Year);
+            data.NumberYears = data.YearlyBudget.Max(r => r.Year) - data.FirstYear;
+
+            try
+            {
+                var simulation = db.SIMULATIONS
+                    .Include(d => d.INVESTMENTS)
+                    .Include(d => d.YEARLYINVESTMENTs)
+                    .Single(_ => _.SIMULATIONID == data.SimulationId);
+
+                if (simulation != null)
```

Comment:
You can just use the ? to check for null

`simulation?. xyz`

Why would simulation be null here? Can you use SingleOrDefault and get a default value back?

Reply:
Correct, tried but cannot get null. In the case where nothing is found from the query an sql exception is thrown. Got rid of check.